src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferConvertToHexStringOverBitConverter.Fixer.cs

```diff
@@ -0,0 +1,102 @@
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetRequiredSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span, getInnermostNodeForTie: true);
+
+            if (node is null)
+            {
+                return;
+            }
+
+            var semanticModel = await context.Document.GetRequiredSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            var operation = semanticModel.GetOperation(node, context.CancellationToken);
+
+            if (operation is not IInvocationOperation replaceInvocation ||
+                replaceInvocation.Instance is not IInvocationOperation instanceInvocation ||
+                !PreferConvertToHexStringOverBitConverterAnalyzer.RequiredSymbols.TryGetSymbols(semanticModel.Compilation, out var symbols) ||
+                !symbols.TryGetBitConverterToStringInvocation(instanceInvocation, out var bitConverterInvocation, out var toLowerInvocation))
+            {
+                return;
+            }
+
+            var codeAction = CodeAction.Create(
+                MicrosoftNetCoreAnalyzersResources.PreferConvertToHexStringOverBitConverterCodeFixTitle,
+                ReplaceWithConvertToHexStringCall,
+                nameof(MicrosoftNetCoreAnalyzersResources.PreferConvertToHexStringOverBitConverterCodeFixTitle));
+
+            context.RegisterCodeFix(codeAction, context.Diagnostics);
+
+            async Task<Document> ReplaceWithConvertToHexStringCall(CancellationToken cancellationToken)
+            {
+                var editor = await DocumentEditor.CreateAsync(context.Document, cancellationToken).ConfigureAwait(false);
+                var generator = editor.Generator;
+                var bitConverterArgumentsInParameterOrder = bitConverterInvocation.Arguments.GetArgumentsInParameterOrder();
+
+                var convertToHexStringSymbol = bitConverterArgumentsInParameterOrder.Length == 1
+                    ? symbols.ConvertToHexString!
+                    : symbols.ConvertToHexStringStartLength!;
+                var typeExpression = generator.TypeExpressionForStaticMemberAccess(convertToHexStringSymbol.ContainingType);
+                var methodExpression = generator.MemberAccessExpression(typeExpression, convertToHexStringSymbol.Name);
+                var methodInvocation = bitConverterArgumentsInParameterOrder.Length switch
+                {
+                    // BitConverter.ToString(data).Replace("-", "") => Convert.ToHexString(data)
+                    1 => generator.InvocationExpression(methodExpression, bitConverterArgumentsInParameterOrder[0].Value.Syntax),
+                    // BitConverter.ToString(data, start).Replace("-", "") => Convert.ToHexString(data, start, data.Length - start)
+                    2 => generator.InvocationExpression(
```

Comment:
Not sure if we should be more restrictive with the `data` argument in this case, as the fixer changes the code to access it twice.

Reply:
What about converting to span overload instead `Convert.ToHexString(data.AsSpan().Slice(start))`?

Reply:
Fixed in 6db3a8483d0c7684aebd6f008441cf5834931f21.